app/forms/ReferenceNumberFormProvider.scala

```diff
@@ -16,17 +16,20 @@
 
 package forms
 
-import javax.inject.Inject
-
 import forms.mappings.Mappings
 import play.api.data.Form
 
+import javax.inject.Inject
 class ReferenceNumberFormProvider @Inject() extends Mappings {
 
   def apply(): Form[String] =
     Form(
-      "value" -> text("referenceNumber.error.required")
-        .verifying(minLength(2, "referenceNumber.error.length"))
-        .verifying(maxLength(64, "referenceNumber.error.length"))
+      "value" -> text("referenceNumber.error.required").transform[String](_.replaceAll("\\s", ""), identity)
```

Comment:
we could use the existing trimWhitespace method here

Reply:
good to know.. done